app/StyledApp.tsx

```diff
@@ -39,12 +41,41 @@ const EventRouter = () => {
   return <></>;
 };
 
+const getOnLineStatus = () =>
+  typeof navigator !== 'undefined' && typeof navigator.onLine === 'boolean'
+    ? navigator.onLine
+    : true;
+
+const useNavigatorOnLine = () => {
+  const [status, setStatus] = useState(getOnLineStatus());
+
+  useEffect(() => {
+    function setOnline() {
+      setStatus(true);
+    }
+    function setOffline() {
+      setStatus(false);
+    }
+    window.addEventListener('online', setOnline);
+    window.addEventListener('offline', setOffline);
+
+    return () => {
+      window.removeEventListener('online', setOnline);
+      window.removeEventListener('offline', setOffline);
+    };
+  }, [setStatus]);
+
+  return status;
+};
```

Comment:
Can't we move it inside `NoInternetConnection` component?

Reply:
I will check